WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsAlerts.swift

```diff
@@ -16,6 +16,24 @@ final class CardReaderSettingsAlerts {
         from.present(newAlert, animated: true)
     }
 
+    func connectingToReader(from: UIViewController) {
```

Comment:
I am just curious, would it be possible to update the model without dismissing it?  It seems like there are maybe too many presentations and dismissals. 

In theory, we could do something like:
```
        let viewModel = connectingToReader()
       if modalController != nil {
          modalComtroller.setViewModel(viewModel)
      }
```

That would update the modal being currently presented without dismissing it first.

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/4172/commits/bd40e9a748889e2eae038e3e342b2d66290d4967 - looks better without those extra animations